### PR TITLE
Add helper CLI-script for Cypress custom-reporter

### PR DESCRIPTION
**Pull request by tlei123**
## Description

Add Node-CLI helper script to facilitate using cy-tr-reporter, which was [recently added](https://github.com/department-of-veterans-affairs/vets-website/pull/18936).

## Original issue(s)
\[N/A\]

## Testing done

- This utility-script does not affect any VA.gov app functionality.
- Tested locally against cy-tr-reporter.  Cypress test-results [successfully pushed to TestRail](https://dsvavsp.testrail.io/index.php?/runs/overview/14) \[see Wednesday, October 7, 2021\].

## Acceptance criteria
- Local testing by Reviewer also succeeded:
  - [x] Test-results successfully pushed to TestRail
  - [x] Generated custom config files not tracked (git-ignored &mdash; not displayed as Untracked).

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/cypress-testrail-helper/lib/utils.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      console.log(chalk.green(`stdout: ${stdout}`));
+    });
+  },
+
+  async getSetAppProjectConfig(myConfig) {
+    let projectAnswers = undefined;
+    let myNewConfig = undefined;
+
+    projectAnswers = await inquirer.askTestRailProjectOptions();
+    // eslint-disable-next-line no-unused-vars, no-param-reassign, no-multi-assign
```
[the remark this answers is not in the document]
Comment (tlei123) [reply script/cypress-testrail-helper/lib/utils.js:104]: \[same as above\]

File: script/cypress-testrail-helper/index.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+projectSwitching =
+  (appArgs.includes('--switch-project') || appArgs.includes('--sp')) &&
+  fs.existsSync(myConfigPath);
+
+// CHECK FOR CONFIG-FILE, GET/SET CONFIG(S), THEN RUN CYPRESS SPEC.
+// IF no config-file exists, create one.
+// IF config-file does exist, create run-specific Cypress config-file.
+switch (fs.existsSync(myConfigPath)) {
+  case true:
+    // eslint-disable-next-line import/no-unresolved
```
[the remark this answers is not in the document]
Comment (tlei123) [reply script/cypress-testrail-helper/index.js:62]: \[Similar to above, but change suggestions welcomed!\]

File: script/cypress-testrail-helper/index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          chalk.red(
+            `'ERROR: ${err}\n[See the app README.md file for instructions.]'`,
+          ),
+        );
+      });
+    break;
+  default:
+    break;
+}
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply script/cypress-testrail-helper/index.js:110]: \[Same as [1st console-log eslint-disable](https://github.com/department-of-veterans-affairs/vets-website/pull/19024#discussion_r723512752).\]

**Comment by tlei123** [comment]
@pjhill @ddzz , about those eslint-disables, I have a couple of ideas:
- Add the script-files to .eslintignore; or
- Make this a separate, 3rd-party NPM-package, similar to what I did for cy-tr-reporter. 

Please let me know your thoughts.  I think va-vfs-bot might be tripping over these disables with future PRs, even if we let this one through as-is.

**Review by pjhill — APPROVED** [review]
Pretty cool, thanks for cranking on this Tze. I gave it a test run using my go-to project [here](https://dsvavsp.testrail.io/index.php?/runs/view/2395&group_by=cases:section_id&group_order=asc).

This looks good and is pretty easy to use!

File: script/cypress-testrail-helper/index.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        chalk.green(
+          'CLI-prompt will return after Cypress child-process exits.\nTHANKS FOR USING CYPRESS-TESTRAIL-HELPER!',
+        ),
+      );
+
+      // Run Cypress spec
+      utils.runCySpec(specPath);
+    })
+    .catch(err => {
+      console.error(chalk.red(`ERROR: ${err}`));
```
Comment (ddzz) [inline script/cypress-testrail-helper/index.js:34]: We might want to add `process.exitCode = 1` here. `console.error` logs to `stderr`, but it does not set an exit code.

File: script/cypress-testrail-helper/index.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+console.log(
+  chalk.green(
+    figlet.textSync('Cypress-TestRail-Helper', {
+      horizontalLayout: 'universal smushing',
+      width: 80,
+    }),
+  ),
+);
+
+// console.log('process.argv:', process.argv);
```
Comment (oseasmoran73) [inline script/cypress-testrail-helper/index.js:50]: Are these still relevant? If not, can we delete them?

File: script/cypress-testrail-helper/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (oseasmoran73) [reply script/cypress-testrail-helper/index.js:111]: Is it not disabled for the entire file? I would imagine that it is given it was declared at the top of the page?

File: script/cypress-testrail-helper/lib/utils.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        );
+        myResolve('succeeded');
+      } catch (err) {
+        myReject('failed');
+      }
+    });
+  },
+};
+
+/* eslint-enable no-console */
```
Comment (oseasmoran73) [inline script/cypress-testrail-helper/lib/utils.js:121]: x2

**Review by oseasmoran73 — APPROVED** [review]

File: script/cypress-testrail-helper/index.js (hunk 5 above)
Comment (tlei123) [reply script/cypress-testrail-helper/index.js:50]: Yes, still relevant as I'll be making incremental changes based on post-launch user-feedback, and this saves me from re-inserting the same console-logs for testing.  Will definitely delete once the code's stable.

**Pull request merged**